node/tests/swamp/swamp.go

```diff
@@ -0,0 +1,169 @@
+func (s *Swamp) createPeer(ks keystore.Keystore) host.Host {
+	key, err := p2p.Key(ks)
+	require.NoError(s.t, err)
+
+	// IPv6 will be starting with 100:0
+	token := make([]byte, 12)
+	rand.Read(token) //nolint:gosec
+	ip := append(net.IP{}, blackholeIP6...)
+	copy(ip[net.IPv6len-len(token):], token)
+
+	// reference to GenPeer func in libp2p/p2p/net/mock/mock_net.go
+	// on how we generate new multiaddr for new peer
+	a, err := ma.NewMultiaddr(fmt.Sprintf("/ip6/%s/tcp/4242", ip))
+	require.NoError(s.t, err)
+
+	host, err := s.Network.AddPeer(key, a)
+	require.NoError(s.t, err)
+
+	return host
+}
+
+// getTrustedHash is needed for celestia nodes to get the trustedhash
+// from CoreClient. This is required to initialize and start correctly.
+func (s *Swamp) getTrustedHash(ctx context.Context) string {
+	bf := core.NewBlockFetcher(s.CoreClient)
+	blocks, err := bf.SubscribeNewBlockEvent(ctx)
+	require.NoError(s.t, err)
+	block := <-blocks
+	hstr := block.Hash().String()
+	err = bf.UnsubscribeNewBlockEvent(ctx)
+	require.NoError(s.t, err)
+	return hstr
+}
+
+// TODO(@Bidon15): CoreClient(limitation)
+// Now, we are making an assumption that consensus mechanism is already tested out
+// so, we are not creating bridge nodes with each one containing its own core client
+// instead we are assigning all created BNs to 1 Core from the swamp
+
+// NewBridgeNode creates a new instance of BridgeNodes. Afterwards,
+// the instance is store in the swamp's BridgeNodes slice.
+func (s *Swamp) NewBridgeNode(options ...node.Option) *node.Node {
+	cfg := node.DefaultConfig(node.Bridge)
+	store := node.MockStore(s.t, cfg)
+
+	ks, err := store.Keystore()
+	require.NoError(s.t, err)
+
+	// TODO(@Bidon15): If for some reason, we receive one of existing options
+	// like <core, host, hash> from the test case, we need to check them and not use
+	// default that are set here
+	options = append(options,
+		node.WithCoreClient(s.CoreClient),
+		node.WithHost(s.createPeer(ks)),
+		node.WithTrustedHash(s.trustedHash),
+	)
+
+	node, err := node.New(node.Bridge, store, options...)
+	require.NoError(s.t, err)
+	s.BridgeNodes = append(s.BridgeNodes, node)
+	return node
+}
+
+// NewLightClient creates a new instance of LightClient. Afterwards,
+// the instance is store in the swamp's LightClients slice
+func (s *Swamp) NewLightClient(options ...node.Option) *node.Node {
+	cfg := node.DefaultConfig(node.Light)
+	store := node.MockStore(s.t, cfg)
+
+	ks, err := store.Keystore()
+	require.NoError(s.t, err)
+
+	// TODO(@Bidon15): If for some reason, we receive one of existing options
+	// like <core, host, hash> from the test case, we need to check them and not use
+	// default that are set here
+	options = append(options,
+		node.WithHost(s.createPeer(ks)),
+		node.WithTrustedHash(s.trustedHash),
+	)
+
+	node, err := node.New(node.Light, store, options...)
+	require.NoError(s.t, err)
+	s.LightClients = append(s.LightClients, node)
+	return node
```

Comment:
You don't Stop Nodes in tests and here, but we need to be nice and close them to test the whole lifecycle as well. Doing t.Cleanup calling Stop of each created Node type would be sufficient. This way we also remove the need for tests to call Stop manually each time. We can also consider returning only Started nodes on NewAnyNode

Reply:
Thanks for pointing this. Done